### PR TITLE
Storage.hh: Let `NodeInfo`, `DirHandle` `FileHandle` be value-comparable

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

File: payload/sp/storage/Storage.hh [hunk 1]
```diff
@@ -78,6 +80,8 @@ public:
     DirHandle &operator=(DirHandle &&);
     ~DirHandle();
 
+    bool operator==(const DirHandle &) const = default;
+
```
Comment (stblr) [inline payload/sp/storage/Storage.hh:84]: What about `FileHandle`?

File: payload/sp/storage/Storage.hh [hunk 2]
```diff
@@ -28,6 +28,8 @@ struct NodeInfo {
     u64 size;
     OSTime tick = 0;
     wchar_t name[255 + 1];
+
+    bool operator==(const NodeInfo &) const = default;
```
Comment (stblr) [inline payload/sp/storage/Storage.hh:32]: What's the default comparison for the `name` array?

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/sp/storage/Storage.hh (hunk 2 above)
Comment (riidefi) [reply payload/sp/storage/Storage.hh:32]: `memcmp`

Comment (riidefi) [reply payload/sp/storage/Storage.hh:32]: Changed to `!wcsncmp`

File: payload/sp/storage/Storage.hh (hunk 1 above)
Comment (riidefi) [reply payload/sp/storage/Storage.hh:92]: You're right, added.

**Review by stblr — APPROVED** [review]

**Pull request merged**